permission/db.go

```
@@ -6,25 +6,42 @@ import (
 )
 
 // getPermissions returns all permissions
-func (s *Store) getPermissions(ctx context.Context) (p []Permission, err error) {
-	err = s.db.SelectContext(ctx, &p, `SELECT *
-		FROM people.permissions
-		ORDER BY name;`)
+func (s *Store) getPermissions(ctx context.Context) ([]Permission, error) {
+	var p []Permission
+	err := s.db.SelectContext(ctx, &p, `SELECT p.*, COUNT(rp.role_id) AS roles
+		FROM people.permissions p
+		LEFT JOIN people.role_permissions rp on p.permission_id = rp.permission_id
+		GROUP BY p, p.permission_id, name, description
+		ORDER BY p.name;`)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get permissions: %w", err)
 	}
 	return p, nil
 }
 
 // getPermissions returns all permissions
-func (s *Store) getPermission(ctx context.Context, id int) (Permission, error) {
-	p := Permission{}
-	err := s.db.GetContext(ctx, &p, `SELECT *
-		FROM people.permissions
-		WHERE permission_id = $1
-		LIMIT 1;`, id)
+func (s *Store) getPermission(ctx context.Context, p1 Permission) (Permission, error) {
+	var p Permission
+	err := s.db.GetContext(ctx, &p, `SELECT p.*, COUNT(rp.role_id) AS roles
+		FROM people.permissions p
+		LEFT JOIN people.role_permissions rp on p.permission_id = rp.permission_id
+		WHERE p.permission_id = $1
+		GROUP BY p, p.permission_id, name, description
+		LIMIT 1;`, p1.PermissionID)
 	if err != nil {
 		return Permission{}, fmt.Errorf("failed to get permission: %w", err)
 	}
 	return p, nil
 }
+
+func (s *Store) addPermission(ctx context.Context, p1 Permission) (Permission, error) {
+	return Permission{}, nil
+}
+
+func (s *Store) editPermission(ctx context.Context, p1 Permission) (Permission, error) {
+	return Permission{}, nil
+}
+
+func (s *Store) deletePermission(ctx context.Context, p1 Permission) error {
+	return nil
+}
```

Should these be filled in? If they're not yet implemented stubs, might be better to panic rather than silently doing nothing.

---

They panic now